### ISSUE TITLE
2.0.10 backport candidates

### ISSUE THREAD

**Issue by arnetheduck** [nim-lang/backport#5]
```
git log --oneline --pretty="format:%m https://github.com/nim-lang/Nim/commit/%H %s" --left-right  --cherry-pick origin/devel...origin/version-2-0 
```

* [x] https://github.com/nim-lang/Nim/commit/770f8d551372893ed480550062a89e76c60ff4a8 opensym for templates + move behavior of opensymchoice to itself (#24007)
* [x] https://github.com/nim-lang/Nim/commit/4ef06a5cc56f6ef32b8ac1e7c11592597406486a fixes `cast` expressions introduces unnecessary copies (#24004)
* [x] https://github.com/nim-lang/Nim/commit/dda638c1ba985a77eac3c7518138992521884172 fixes #23945; type checking for whenvm expresssions (#23970)
* [x] https://github.com/nim-lang/Nim/commit/34719cad9d95b9a13fbbc5bfb9e5e06662a8c6ed allow `untyped` arguments to fail to compile in overload mismatch error (#23984)
* [x] https://github.com/nim-lang/Nim/commit/58813a3b2e375d91a307645168765372fb4584c8 make all generic aliases tyAlias (#23978)
* [x] https://github.com/nim-lang/Nim/commit/f7c11a8978a1fc7182ef18c4bdc80e920ce6ad88 allow generic compileTime proc folding (#22022)
* [x] https://github.com/nim-lang/Nim/commit/995081b56a098c431eb24fbb37afadb4cee2fb4b fix `is` with `type`/`typedesc` crashing the compiler (#23967)
* [x] https://github.com/nim-lang/Nim/commit/298ada3412c9cf5971abc2b3b891b9bb8612e170 fixes #23954; uint8 > 8 bit at compile-time (#23955)
* [x] https://github.com/nim-lang/Nim/commit/c5b206d4ac6704147d9ec27506b9d0171f3a9501 fix #23817; Use __builtin_saddl_overflow variants for arm-none-eabi-gcc. (#23835)
* [x] https://github.com/nim-lang/Nim/commit/f0e1eef65ed08fb153b88615f07ea0fd91e4c258 fixes #14522 #22085 #12700 #23132; no range check for uints (#23930)
* [x] https://github.com/nim-lang/Nim/commit/bd063113ec3c00d80fc32a5581ac9552682f6b43 fixes #23894; succ/pred shouldn't raise OverflowDefect for unsigned integers (#23895)
* [x] https://github.com/nim-lang/Nim/commit/925dc5c1319bce935593cc12380feec1538e2ce5 fixes #19171; have `openArray` converted from `ptr UncheckedArray` be mutable (#23882)
* [x] https://github.com/nim-lang/Nim/commit/97f54745459b8651b7a38c174b3a8135224ebd09 fix generics treating symchoice symbols as uninstantiated (#23860)
* [x] https://github.com/nim-lang/Nim/commit/e53a2ed19bf452c2f5db7b691c403a43a26fd5c3 fixes #20865; fixes #20987; Missing bounds check in array slicing (#23814)
* [x] https://github.com/nim-lang/Nim/commit/288d5c4ac32c27704258a15d365a56dadf38b0e9 fixes #5091; Ensure we don't wait on an exited process on Linux (#23743)
* [x] https://github.com/nim-lang/Nim/commit/2a658c64d8c24ad4a0944e01ba3e005bfb12534f fixes #23732, os.sleep(-1) now returns immediately (#23734)
* [ ] https://github.com/nim-lang/Nim/commit/ae4b47c5bd48d244ee1f93ec6ba5f6bcf55eb973 fixes #20048; fixes #15746; don't sink object fields if it's of openarray type (#23608)
* [x] https://github.com/nim-lang/Nim/commit/948bb38335985c2c873612e66e2e054ef0573b58 ref #20653; fixes chronos empty case branches (#23706)
* [x] https://github.com/nim-lang/Nim/commit/8037bbe327ec581fbafd558bf1eb4619d373e7ec Fix non-exported `memfiles.setFileSize` to be able to shrink files on posix via `memfiles.resize` (#23717)
* [ ] https://github.com/nim-lang/Nim/commit/8cbbe12ee4eaa9568d246638b1f712a5a57e20ab fixes #22398; [backport] (#23704)
* [x] https://github.com/nim-lang/Nim/commit/56c95758b2777a36c8608a7b9950ba5ce8ce0b65 fixes #23445; fixes #23418 [backport] (#23699)
* [x] https://github.com/nim-lang/Nim/commit/de4c7dfdd9501e286e0e525446e82ff340a628f4 fixes #22798; Duplicate libraries linker warning (i.e., '-lm') on macOS (#23292)
* ~~[ ] https://github.com/nim-lang/Nim/commit/d923c581c118b9ea891785bbb828c3cdede587b4 revert #23436; remove workaround (#23653)~~ not applicable to v2.0.x
* [x] https://github.com/nim-lang/Nim/commit/cc5ce72376872c7080543593ee5e7234b16a62bd fixes #23635; tasks.toTask Doesn't Expect a Dot Expression (#23641)
* [x] https://github.com/nim-lang/Nim/commit/c615828ccb2b309ac03aab93d94b40af6b4079ff fixes #22852; fixes #23435; fixes #23645; SIGSEGV when slicing string or seq[T] with index out of range (#23279)
* [x] https://github.com/nim-lang/Nim/commit/3bda5fc840870f6589e4bdbcbbdc811bde599969 Handle arbitrarily long symlink target in `expandSymlinks()` (#23650)
* [x] https://github.com/nim-lang/Nim/commit/5cd141cebb3ad7504ec399a9b7140fed40049bf1 fixes `reifiedOpenArray`; `nkHiddenStdConv` is PathKinds1 not PathKinds0 (#23633)

**Comment by tersec** [comment]
When this is merged into `devel`:
- [x] https://github.com/nim-lang/Nim/pull/24105

**Comment by tersec** [comment]
- [x] https://github.com/nim-lang/Nim/pull/24130 when merged into `devel`

**Comment by tersec** [comment]
- [x] https://github.com/nim-lang/Nim/pull/24152
- [ ] https://github.com/nim-lang/Nim/pull/24166 when merged
- [x] https://github.com/nim-lang/Nim/pull/24170
- [ ] https://github.com/nim-lang/Nim/pull/24189

**Comment by narimiran** [comment]
@tersec Until now (when I saw it mentioned in https://github.com/nim-lang/Nim/pull/24189) I haven't realized that you edited your message above and added that PR to the list.
As such, it is unfortunately not part of 2.0.10, but I'll backport it now and it will be in 2.0.12.

And since 2.0.10 is out, I'm closing this issue.
